Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,335 @@ function Test-DiscoveryService{
+      Write-DosMessage -Level "Verbose" -Message "No AAD Settings found in install.config"
+      return
+    }
+
+    # Gather AAD Settings from install.config
+    $existingChildNodes = Get-XMLChildNodes -installConfigPath $installConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute
+    $azureSecretNameFromInstallConfig = Get-XMLChildNode -installConfigPath $installConfigPath -configSection $configSection -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting "azureSecretName" 
+    $childNodeHeader = @{}
+    $childNodeHeader.Add("secretName","")
+    $existingChildNodes += $childNodeHeader
+    $existingChildNodes += $azureSecretNameFromInstallConfig
+    Write-DosMessage -Level "Verbose" -Message "Copied the AAD Settings from install.config"
+
+    # Remove blank child nodes and azureSecretName from default azuresettings.config manifest
+    Write-DosMessage -Level "Verbose" -Message "Started removing blank child variables and azureSecretName from azuresettings.config in nuget folder"
+    Remove-XMLChildNodes -azureConfigPath $azureConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting ""
+    Remove-XMLChildNode -azureConfigPath $azureConfigPath -configSection $configSection -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting "azureSecretName"
+    Write-DosMessage -Level "Verbose" -Message "Finished removing blank child variables and azureSecretName from azuresettings.config in nuget folder"
+
+    # Add AAD settings to azuresettings.config
+    Write-DosMessage -Level "Verbose" -Message "Started adding child variables to azuresettings.config that were found in install.config"
+    $nodesToSearch += "secretName"
+    Add-XMLChildNodes -azureConfigPath $azureConfigPath -configSection $configSection -childNodesInOrder $nodesToSearch -childNodesToAdd $existingChildNodes
+
+    # Gather AAD Settings from azuresettings.config
+    $existingAzureChildNodes = Get-XMLChildNodes -installConfigPath $azureConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute
+    Write-DosMessage -Level "Verbose" -Message "Copied the AAD Settings from azuresettings.config for comparison"
+
+    # Check if install.config and azuresettings.config AAD settings match
+    $areSameGroup = Compare-Object -ReferenceObject $existingChildNodes -DifferenceObject $existingAzureChildNodes -Property name, value, alias, appName, tenantId, tenantAlias, clientid, secret
+    Write-DosMessage -Level "Verbose" -Message "Comparing the AAD Settings in install.config and azuresettings.config, before deleting from install.config"
+
+    if($null -ne $areSameGroup)
+    {
+      Write-DosMessage -Level "Verbose" -Message "Since install.config and azuresettings.config AAD Settings are different, install.config settings were not deleted"
```

Comment:
Probably want this line in "Informational" or higher so it comes up even when not debugging, since it sounds like something went wrong somewhere

Reply:
changed